pkgs/development/libraries/goocanvas/2.x.nix

```
@@ -41,6 +42,16 @@ stdenv.mkDerivation rec {
     glib
   ];
 
+  # add fedora patch to fix gcc-14 build
+  # https://src.fedoraproject.org/rpms/goocanvas2/tree/main
+  patches = [
+    (fetchpatch rec {
+      name = "goocanvas-2.0.4-Fix-building-with-GCC-14.patch";
+      hash = "sha256-9uqqC1uKZF9TDz5dfDTKSRCmjEiuvqkLnZ9w6U+q2TI=";
+      url = "https://src.fedoraproject.org/rpms/goocanvas2/raw/e799612a277262a0c6bd03db10a6ee9ca7871b9c/f/${name}";
+    })
```

```suggestion
    (fetchpatch {
      name = "goocanvas-2.0.4-Fix-building-with-GCC-14.patch";
      url = "https://src.fedoraproject.org/rpms/goocanvas2/raw/e799612a277262a0c6bd03db10a6ee9ca7871b9c/f/goocanvas-2.0.4-Fix-building-with-GCC-14.patch";
      hash = "sha256-9uqqC1uKZF9TDz5dfDTKSRCmjEiuvqkLnZ9w6U+q2TI=";
    })
```

Using the full URL seems to be preferred (there're 2600+ `fetchpatch {` versus 5 `fetchpatch rec {` in the tree). LGTM otherwise.